cmd/podman/ps.go

```diff
@@ -71,12 +72,72 @@ type psJSONParams struct {
 	Namespaces       *batchcontainer.Namespace `json:"namespace,omitempty"`
 }
 
-// Type declaration and functions for sorting the PS output by time
+// Type declaration and functions for sorting the PS output
 type psSorted []psTemplateParams
 
-func (a psSorted) Len() int           { return len(a) }
-func (a psSorted) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a psSorted) Less(i, j int) bool { return a[i].CreatedAtTime.After(a[j].CreatedAtTime) }
+func (a psSorted) Len() int      { return len(a) }
+func (a psSorted) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+type psSortedCommand struct{ psSorted }
+
+func (a psSortedCommand) Less(i, j int) bool { return a.psSorted[i].Command < a.psSorted[j].Command }
+
+type psSortedCreated struct{ psSorted }
+
+func (a psSortedCreated) Less(i, j int) bool {
+	return a.psSorted[i].CreatedAtTime.After(a.psSorted[j].CreatedAtTime)
+}
+
+type psSortedId struct{ psSorted }
+
+func (a psSortedId) Less(i, j int) bool { return a.psSorted[i].ID < a.psSorted[j].ID }
+
+type psSortedImage struct{ psSorted }
+
+func (a psSortedImage) Less(i, j int) bool { return a.psSorted[i].Image < a.psSorted[j].Image }
+
+type psSortedLabels struct{ psSorted }
+
+func (a psSortedLabels) Less(i, j int) bool { return a.psSorted[i].Labels < a.psSorted[j].Labels }
+
+type psSortedMounts struct{ psSorted }
+
+func (a psSortedMounts) Less(i, j int) bool { return a.psSorted[i].Mounts < a.psSorted[j].Mounts }
+
+type psSortedNames struct{ psSorted }
+
+func (a psSortedNames) Less(i, j int) bool { return a.psSorted[i].Names < a.psSorted[j].Names }
+
+type psSortedPorts struct{ psSorted }
+
+func (a psSortedPorts) Less(i, j int) bool { return a.psSorted[i].Ports < a.psSorted[j].Ports }
+
+type psSortedRunningFor struct{ psSorted }
+
+func (a psSortedRunningFor) Less(i, j int) bool {
+	return a.psSorted[i].RunningFor < a.psSorted[j].RunningFor
+}
+
+type psSortedStatus struct{ psSorted }
+
+func (a psSortedStatus) Less(i, j int) bool { return a.psSorted[i].Status < a.psSorted[j].Status }
+
+type psSortedSize struct{ psSorted }
+
+func (a psSortedSize) Less(i, j int) bool {
```

Comment:
Will this still work if `--size` is not specified?

Reply:
It seems it will not. good catch. Back to the drawing boards!

Reply:
now it will!